fboss/agent/hw/sai/SaiSwitch.cpp

```diff
@@ -0,0 +1,1204 @@
+/*
+ * Copyright (c) 2004-present, Facebook, Inc.
+ * Copyright (c) 2016, Cavium, Inc.
+ * All rights reserved.
+ * 
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ * 
+ */
+#include <cstdlib>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/file.h>
+#include <fcntl.h>
+
+#include <boost/foreach.hpp>
+#include <folly/IPAddress.h>
+#include <folly/io/Cursor.h>  // TODO: Remove this when AddHostsForDemo are removed
+
+#include "SaiSwitch.h"
+#include "SaiVrfTable.h"
+#include "SaiNextHopTable.h"
+#include "SaiHostTable.h"
+#include "SaiHost.h"
+#include "SaiRxPacket.h"
+#include "SaiTxPacket.h"
+#include "SaiPortTable.h"
+#include "SaiPortBase.h"
+#include "SaiIntfTable.h"
+#include "SaiRouteTable.h"
+#include "SaiWarmBootCache.h"
+#include "SaiError.h"
+#include "SaiIntf.h"
+
+#include "fboss/agent/packet/ArpHdr.h" // TODO: Remove this when AddHostsForDemo are removed
+#include "fboss/agent/ArpHandler.h"    // TODO: Remove this when AddHostsForDemo are removed
+#include "fboss/agent/state/ArpEntry.h"
+#include "fboss/agent/state/DeltaFunctions.h"
+#include "fboss/agent/state/Interface.h"
+#include "fboss/agent/state/InterfaceMap.h"
+#include "fboss/agent/state/NodeMapDelta.h"
+#include "fboss/agent/state/Port.h"
+#include "fboss/agent/state/PortMap.h"
+#include "fboss/agent/state/StateDelta.h"
+#include "fboss/agent/state/SwitchState.h"
+#include "fboss/agent/state/Vlan.h"
+#include "fboss/agent/state/VlanMap.h"
+#include "fboss/agent/state/VlanMapDelta.h"
+#include "fboss/agent/state/Route.h"
+#include "fboss/agent/state/RouteTableRib.h"
+#include "fboss/agent/state/RouteTable.h"
+#include "fboss/agent/state/RouteTableMap.h"
+#include "fboss/agent/state/RouteDelta.h"
+
+#include "fboss/agent/SwitchStats.h"
+
+using std::make_shared;
+using std::shared_ptr;
+using std::unique_ptr;
+using std::string;
+
+using folly::make_unique;
+using folly::IPAddress;
+
+using facebook::fboss::DeltaFunctions::forEachChanged;
+using facebook::fboss::DeltaFunctions::forEachAdded;
+using facebook::fboss::DeltaFunctions::forEachRemoved;
+
+
+//extern "C" {
```

Comment:
Thanks @oridb 
In fact there are two blocks of code I am planning to uncomment in this file : 
line #71 and #101 . (we will enable them very soon). There is a pending pull request in SAI that needs to be addressed. https://github.com/opencomputeproject/SAI/pull/33

Cavium's SAI adapter requires this and hence i'd like to keep it as a place holder. Is there a different way to mention that this is a TBD code ?